### ISSUE TITLE
Invalid meson arg in mesa formula

### ISSUE THREAD

**Issue by jclRatepay** [Homebrew/homebrew-core#88254]
### `brew gist-logs <formula>` link OR `brew config` AND `brew doctor` output

```shell
HOMEBREW_VERSION: 3.3.1
ORIGIN: https://github.com/Homebrew/brew
HEAD: 5889b44a6031af91b14ee7c7805e64d1b10ad51c
Last commit: 4 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 0fd61678fc69491440bc1428441b9f54eabeb0cf
Core tap last commit: 2 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /home/xxx/.linuxbrew
HOMEBREW_REPOSITORY: /home/xxx/.linuxbrew/Homebrew
HOMEBREW_CELLAR: /home/xxx/.linuxbrew/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_DISPLAY: :0
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_NO_AUTO_UPDATE: set
Homebrew Ruby: 2.6.8 => /home/xxx/.linuxbrew/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.8/bin/ruby
CPU: octa-core 64-bit skylake
Clang: 13.0.0
Git: 2.17.1 => /usr/bin/git
Curl: 7.58.0 => /usr/bin/curl
Kernel: Linux 4.18.20-041820-generic x86_64 GNU/Linux
OS: Ubuntu 18.04.6 LTS (bionic)
Host glibc: 2.27
/usr/bin/gcc: 7.5.0
/usr/bin/ruby: 2.5.1
glibc: N/A
gcc@5: N/A
xorg: N/A

Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  freeglut
  awscli
  netpbm
  terraform
  mesa-glu
  mesa
  jasper

Warning: You have uncommitted modifications to Homebrew/homebrew-core.
If this is a surprise to you, then you should stash these modifications.
Stashing returns Homebrew to a pristine state but can be undone
should you later need to do so for some reason.
  cd /home/jclausen/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core && git stash && git clean -d -f

Uncommitted files:
   M Formula/mesa.rb

Warning: Your Homebrew's prefix is not /home/linuxbrew/.linuxbrew.
Some of Homebrew's bottles (binary packages) can only be used with the default
prefix (/home/linuxbrew/.linuxbrew).
You will encounter build failures with some formulae.
Please create pull requests instead of asking for help on Homebrew's GitHub,
Twitter or any other official channels. You are responsible for resolving
any issues you experience while you are running this
unsupported configuration.
```

### 

- [X] I ran `brew update` and am still able to reproduce my issue.
- [X] I have resolved all warnings from `brew doctor` and that did not fix my problem.

### What were you trying to do (and why)?

Running 'brew install awscli' to install awscli on my Ubuntu machine

### What happened (include all command output)?

Brew resolved a bunch of required dependencies of whom one was the formula called "mesa". After downloading the latest version of all required formulars homebrew failed installing mesa with the following output:

==> Installing dependencies for awscli: mesa, mesa-glu, freeglut, jasper and netpbm
==> Installing awscli dependency: mesa
==> python3 -m venv --system-site-packages /home/jclausen/.linuxbrew/Cellar/mesa/21.2.4/libexec/venv
==> /home/jclausen/.linuxbrew/Cellar/mesa/21.2.4/libexec/venv/bin/pip install -v --no-deps --no-binary :all: --ignore-installed /tmp/mesa--Mako-20211029-3134-1jn07mc/Mako-1.1.5
==> meson --prefix=/home/jclausen/.linuxbrew/Cellar/mesa/21.2.4 --libdir=/home/jclausen/.linuxbrew/Cellar/mesa/21.2.4/lib --buildtype=release --wrap-mode=nofallback .. -Db_ndebug=true -Dplatforms=x11,wayl
Last 15 lines from /home/jclausen/.cache/Homebrew/Logs/mesa/03.meson:
-Dgles2=true
-Dxvmc=true
-Dvalgrind=false
-Dtools=drm-shim,etnaviv,freedreno,glsl,nir,nouveau,xvmc,lima

The Meson build system
Version: 0.60.0
Source dir: /tmp/mesa-20211029-3134-41zq60/mesa-21.2.4
Build dir: /tmp/mesa-20211029-3134-41zq60/mesa-21.2.4/build
Build type: native build
Program python found: YES (/home/jclausen/.linuxbrew/Cellar/mesa/21.2.4/libexec/venv/bin/python)

../meson.build:21:0: ERROR: Unknown options: "xvmc"

A full log can be found at /tmp/mesa-20211029-3134-41zq60/mesa-21.2.4/build/meson-logs/meson-log.txt

READ THIS: https://docs.brew.sh/Troubleshooting

### What did you expect to happen?

Installing awscli and further required formulars without error.

### Step-by-step reproduction instructions (by running `brew` commands)

```shell
brew install awscli (version 2.3.2)

Please note: I already investigatet what exactly causes this error and it turns out, that when looking in the "mesa@21.2.4" formula the "meson" build tool is invoked with the option "Dxvmc=true" (which is unknown related to the meson output). When looking at the downloaded mesa-21.2.4 resources the related option has a different name, which is "gallium-xvmc". So instead of calling meson with "-Dxvmc=true" it needs to be "Dgallium-xvmc=true". Tryed it on my machine in both directions (once changed the supported option's name in the mesa sources and also correcting the option name in the homebrew mesa formula) and both times it worked as expected.
```

**Comment by carlocab** [comment]
Thanks for the report. Do you mind opening a pull request to fix this?

**Comment by jclRatepay** [comment]
Pull request has been opened: https://github.com/Homebrew/homebrew-core/pull/88496

**Comment by Moisan** [comment]
Fixed by #88496.
